src/Resolver/InstanceCreator.php

```diff
@@ -4,7 +4,11 @@
 
 namespace Gacela\Resolver;
 
-final class InstanceCreator
+use Psr\Container\ContainerInterface;
+
+use function is_object;
+
+final class InstanceCreator implements ContainerInterface
```

Comment:
The interface name from the PSR people is called `ContainerInterface`, should we rename this class from `InstanceCreator` to just `Container`?
@Chemaclass @Tito-Kati